x/concentrated-liquidity/swaps_test.go

```diff
@@ -85,66 +85,120 @@ var (
 			tokenOutDenom: "eth",
 			priceLimit:    sdk.NewDec(5004),
 			spreadFactor:  sdk.ZeroDec(),
```

Comment:
Note to reviewer: the majority of the diff is in this file. Not sure if it's time efficient to manually review all the changes, but all the python scripts can be directly tested in an interactive python shell with the following command:
```
python3 -i x/concentrated-liquidity/python/swap_test.py
```